### PR TITLE
mh_sha1: Implement multi-hash SHA1 with crypto extension

### PR DESCRIPTION
**Pull request by haoyu01**
This is mh_sha1 with crypto extension.

### CODE REVIEW

**Comment by haoyu01** [comment]
@gbtucker , please review

File: mh_sha1/mh_sha1_internal.h [hunk 1]
```diff
@@ -316,6 +316,21 @@
...
+  * @param  input_data Pointer to input data to be processed
+  * @param  digests 16 segments digests
+  * @param  frame_buffer Pointer to buffer which is a temp working area
+  * @param  num_blocks The number of blocks.
+  * @returns none
+  *
+  */
+ void mh_sha1_block_ce(const uint8_t * input_data, uint32_t digests[SHA1_DIGEST_WORDS][HASH_SEGS],
+			 uint8_t frame_buffer[MH_SHA1_BLOCK_SIZE], uint32_t num_blocks);
+
```
Comment (gbtucker) [inline mh_sha1/mh_sha1_internal.h]: Can we move this to the `mh_sha1/aarch64/` directory so we keep the arch-specific parts separate?  Could just put the declaration in `aarch64/mh_sha1_ce.c` for now or have its own internal header.

**Comment by haoyu01** [comment]
@gbtucker  , just remove the function from mh_sha1_internal.h . Please review

**Comment by haoyu01** [comment]
@gbtucker , updated . please review .

**Pull request merged**